common/dynamicconfig/constants.go

```diff
@@ -428,6 +428,8 @@ const (
 	MatchingLongPollExpirationInterval = "matching.longPollExpirationInterval"
 	// MatchingSyncMatchWaitDuration is to wait time for sync match
 	MatchingSyncMatchWaitDuration = "matching.syncMatchWaitDuration"
+	// MatchingHistoryMaxPageSize is the maximum page size of history events returned on PollWorkflowTaskQueue requests
+	MatchingHistoryMaxPageSize = "matching.historyMaxPageSize"
```

Comment:
nit: I remember there's a config used by history service for controlling the same thing? Can we reuse that? I feel like now we have one such config for each service (frontend, history & matching) but they are really the same thing, and probably don't need separate control.

Reply:
Since history and frontend already had separate options, I thought it would be clearer to add a matching specific one. I agree though, it really is a global setting and we don't need those separate ones. I don't have a strong preference on reusing vs adding a new one though.